pytest_plugins/requirements/req_updater.py

```diff
@@ -0,0 +1,55 @@
+import subprocess
+from pathlib import Path
+from functools import cached_property
+
+class ReqUpdater:
+
+    # Installed package name as key and its counterpart in requirements file as value
+    package_deviates = {'Betelgeuse': 'betelgeuse', 'broker': 'broker[docker]', 'dynaconf': 'dynaconf[vault]'}
+
+    @cached_property
+    def installed_packages(self):
+        """Returns the list of installed packages in venv
+
+        This also normalizes any package names that deviates in requirents file vs installed names
+        """
+        installed_pkges = str(subprocess.run('pip list --format=freeze', capture_output=True, shell=True).stdout.decode())
+        for pkg in self.package_deviates:
+            if pkg in installed_pkges:
+                # Replacing the installed package names with their names in requirements file
+                installed_pkges = installed_pkges.replace(pkg, self.package_deviates[pkg])
+        return installed_pkges.splitlines()
+
+    def package_filter(self, package):
+        discard_list = ('#', 'git', '--')
+        return not package.startswith(discard_list)
+
+    @cached_property
+    def requirements_packages(self):
+        req_packages = Path('requirements.txt').read_text().replace('\n\n', '\n').splitlines()
+        return list(filter(self.package_filter, req_packages))
+
+    @cached_property
+    def optional_packages(self):
+        opt_packages = Path('requirements-optional.txt').read_text().replace('\n\n', '\n').splitlines()
```

Comment:
```python
        with open('requirements-optional.txt') as f:
            opt_packages = [line.strip() for line in f.readlines() if line.strip()]
```
instead of `\n\n` using strip would make sense